tests/virt_autotest/kubevirt_tests_server.pm

```diff
@@ -0,0 +1,372 @@
+    assert_script_run("curl " . data_url("virt_autotest/kubevirt_tests/$test_conf") . " -o $test_conf");
+    assert_script_run("curl " . data_url("virt_autotest/kubevirt_tests/$parser_script") . " -o $parser_script");
+
+    my $test_suite_config = '/usr/share/kube-virt/manifests/testing/default-config.json';
+    my ($go_test, $skip_test, $extra_opt, $params);
+    my ($artifacts, $junit_xml, $test_log, $test_cmd);
+    my $retry_times = get_required_var('FAILED_RETRY');
+
+    # Workaround for bsc#1199448
+    my $node_helper = 'node-helper.yaml';
+    assert_script_run("curl " . data_url("virt_autotest/kubevirt_tests/$node_helper") . " -o $node_helper");
+    assert_script_run("kubectl apply -f $node_helper");
+
+    foreach my $section (@kubevirt_tests) {
+        $go_test   = '';
+        $skip_test = '';
+        $extra_opt = '';
+        $params    = script_output("perl $parser_script $test_conf $section");
+        ($go_test, $skip_test, $extra_opt) = split(',', $params);
+        record_info($section, join("\n", $go_test, $skip_test, $extra_opt));
+
+        if ($section eq 'operator_test') {
+            my $pre_rel_registry = get_required_var('PREVIOUS_RELEASE_REGISTRY');
+            my $pre_rel_tag = get_required_var('PREVIOUS_RELEASE_TAG');
+            $extra_opt =~ s/PREVIOUS_RELEASE_REGISTRY/$pre_rel_registry/;
+            $extra_opt =~ s/PREVIOUS_RELEASE_TAG/$pre_rel_tag/;
+        }
+
+        $artifacts = "$result_dir/$section";
+        $junit_xml = "$result_dir/${section}.xml";
+        $test_log  = "$result_dir/${section}.log";
+        $test_cmd  = "virt-tests -ginkgo.regexScansFilePath=true ".
+                     "-ginkgo.focus='$go_test' ".
+                     "-ginkgo.skip='QUARANTINE$skip_test' ".
+                     "-ginkgo.slowSpecThreshold 60 ".
+                     "-kubeconfig=/root/.kube/config ".
+                     "-kubectl-path=`which kubectl` ".
+                     "-virtctl-path=`which virtctl` ".
+                     "-installed-namespace=kubevirt ".
+                     "-deploy-testing-infra=false ".
+                     "-config=$test_suite_config ".
+                     "-dns-service-name=rke2-coredns-rke2-coredns ".
+                     "-ginkgo.v=true ".
+                     "-test.v=true ".
+                     "-ginkgo.trace=true ".
+                     "-ginkgo.noisySkippings=false ".
+                     "-ginkgo.progress=true ".
+                     "-ginkgo.noColor ".
+                     "-apply-default-e2e-configuration ".
+                     "$extra_opt ".
+                     "-artifacts=$artifacts ".
+                     "-junit-output=$junit_xml ".
+                     "2>&1 | tee $test_log";
+
+        # my $times;
+        # while ($times && script_run($test_cmd, timeout => 5400)) {
+        #     $times--;
+        # }
+        $retry_times = 1 if ($retry_times);
+        script_retry($test_cmd, retry => $retry_times, delay => 15, timeout => 7200);
```

Comment:
Since some of cases were not very stable, we'd like to rerun it if any case failed within each test section. Besides, each test case has included clean-up processes, so we can just leave its retry interval time with default 30s.